src/fd.cpp

```diff
@@ -85,21 +107,16 @@ static int wait_thread(void *arg) {
     } else if (waitFor_ms > LATER_POLL_INTERVAL) {
       waitFor_ms = LATER_POLL_INTERVAL;
     }
-    ready = LATER_POLL_FUNC(pollfds.data(), args->num_fds, static_cast<int>(waitFor_ms));
+    ready = LATER_POLL_FUNC(args->fds->data(), args->num_fds, static_cast<int>(waitFor_ms));
     if (args->flag->load()) return 1;
     if (ready) break;
   }
 
   // store pollfd revents in args->results for use by callback
+
   if (ready > 0) {
-    for (int i = 0; i < args->rfds; i++) {
-      (*args->results)[i] = pollfds[i].revents == 0 ? 0 : pollfds[i].revents & POLLIN ? 1: R_NaInt;
-    }
-    for (int i = args->rfds; i < (args->rfds + args->wfds); i++) {
-      (*args->results)[i] = pollfds[i].revents == 0 ? 0 : pollfds[i].revents & POLLOUT ? 1 : R_NaInt;
-    }
-    for (int i = args->rfds + args->wfds; i < args->num_fds; i++) {
-      (*args->results)[i] = pollfds[i].revents != 0;
+    for (int i = 0; i < args->num_fds; i++) {
+      (*args->results)[i] = (*args->fds)[i].revents == 0 ? 0 : (*args->fds)[i].revents & (POLLIN | POLLOUT) ? 1: R_NaInt;
```

Comment:
I like that you collapsed this to one for-loop, but I think there is a slight difference in behavior: exceptfds that used to get `TRUE` will now get `NA`.

Reply:
I think this is fine as before it could only be TRUE or FALSE, and now it can only be FALSE or NA for those expectfds. I've already updated the documentation to say that all exceptions produce NAs. It's somewhat more consistent this way.

Reply:
Ah, so you are thinking "`NA` means exception" while I am thinking "`TRUE` means you got what you were waiting for". I was thinking it's nice to be able to say `any(results)` to see if anything succeeded.

In practice I'm guessing none of this matters because 1) exceptfds seems to not really be used in practice these days, 2) probably the correct way to respond to `later_fd` (unless `all(results == FALSE)`) is to go ahead with the read/write, even if there were exceptions, so that you actually get errors. Do you agree?

Reply:
Yes agree - if you get NA then that means "check it" or some sort of action. Only `all(results == FALSE)` would mean no action / try again.

Reply:
Also `any(results)` wouldn't be enough in practice as you'd need to handle the possibility a read or write fd returned `NA`. Unless you're just monitoring except fds, in which case you can just use `anyNA()` instead!